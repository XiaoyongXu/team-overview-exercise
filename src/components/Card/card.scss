.card-root{
  all: unset;
  width: 270px !important;
  height: 280px;
  margin: 20px;
}