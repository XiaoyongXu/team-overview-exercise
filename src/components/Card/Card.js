import React from 'react'
import {
  Card,
  CardImg,
  CardBody,
  CardSubtitle,
  CardTitle,
} from 'reactstrap'
import './card.scss'

const CardItem = ({ style, title, subTitle, index, onClick }) => {
  return (
    <button
      className="card-root"
      type="button"
      onClick={onClick}
      style={style}
      data-testid="card"
    >
      <Card>
        <CardImg top width="100%" src={`https://picsum.photos/id/${index}/260/216`} alt="Card image cap" />
        {(title || subTitle) && <CardBody>
          {title && <CardTitle tag="h5">{title}</CardTitle>}
          {subTitle && <CardSubtitle style={{ fontSize: 12, color: 'gray' }}>{subTitle}</CardSubtitle>}
        </CardBody> }
      </Card>
    </button>
  )
}

export default CardItem