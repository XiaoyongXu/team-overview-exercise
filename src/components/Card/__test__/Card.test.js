import React from 'react'
import ReactDOM from 'react-dom'
import Card from '../Card'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<Card />, div)
})

it("renders Card correctly with title", () => {
  const { getByTestId } = render(<Card title="test title" />)
  expect(getByTestId('card')).toHaveTextContent("test title")
})

it("renders Card correctly with subtitle", () => {
  const { getByTestId } = render(<Card subTitle="test subTitle" />)
  expect(getByTestId('card')).toHaveTextContent("test subTitle")
})

it("matches snapshot 1", () => {
  const tree = renderer.create(<Card title="test title" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot 2", () => {
  const tree = renderer.create(<Card title="test subtitle" />).toJSON()
  expect(tree).toMatchSnapshot()
})