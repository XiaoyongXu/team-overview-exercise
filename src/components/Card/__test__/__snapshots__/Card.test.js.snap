// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="card-root"
  data-testid="card"
  type="button"
>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://picsum.photos/id/undefined/260/216"
      width="100%"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        test title
      </h5>
    </div>
  </div>
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="card-root"
  data-testid="card"
  type="button"
>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://picsum.photos/id/undefined/260/216"
      width="100%"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        test subtitle
      </h5>
    </div>
  </div>
</button>
`;
